### ISSUE TITLE
Support HealthCheckGracePeriodSeconds for Load Balanced Web Services

### ISSUE THREAD

**Issue by mikelhamer** [aws/copilot-cli#2506]
In my manifest I have defined `image.healthcheck.start_period` with a value of 120. But when I look in my ECS service in the details tab it shows this

![image](https://user-images.githubusercontent.com/10012080/123134335-ae1a5200-d41e-11eb-9ae1-546e2ed81ac5.png)

Is this the same setting, or maybe it's just not working?

**Comment by efekarakus** [comment]
Hi @mikelhamer ! Can you try with `120s` as an input? (adding the unit at the end)

**Comment by mikelhamer** [comment]
Hey @efekarakus thanks for the suggestion! Unfortunately this didn't seem to have any effect.

**Comment by efekarakus** [comment]
Hi @mikelhamer, I think this is happening due to a confusion around which healthcheck we are talking about 😄 

## What's happening now?

The `image` healthcheck field configures the container healthcheck. For example:
```yaml
image:
  build: frontend/Dockerfile
  healthcheck:
    command: ["CMD-SHELL", "curl -f http://localhost:8080 || exit 1"]
    start_period: 120s
  port: 8080
```
And I can see the container healtcheck being configured accordingly:
![Screen Shot 2021-06-23 at 12 42 26 PM](https://user-images.githubusercontent.com/879348/123158468-acea2480-d420-11eb-9f8d-396f9195e623.png)
From navigating to the task definition:
![Screen Shot 2021-06-23 at 12 44 46 PM](https://user-images.githubusercontent.com/879348/123158633-e6229480-d420-11eb-870f-9c1ad96dec62.png)

## What do we want?
I believe what you are looking for is configuring [HealthCheckGracePeriodSeconds in the ECS service](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html#cfn-ecs-service-healthcheckgraceperiodseconds) which we don't surface at the moment 🙇 

This would be a feature request for us where we add a `start_period` field under the [`http` section](https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#http) where the default is 60s.

```yaml
http:
  healthcheck:
     start_period: 120s # new field
```

**Comment by efekarakus** [comment]
One way of mitigating this concern I think for now until we surface this field is modifying the application code to have two healthcheck handlers:
1. For the target group healthcheck that always just responds OK
2. For container healthchecks that has the current logic with a `start_period` of 120s.

Then once this field is supported, the healthchecks can be combined and the container healthcheck can be removed

**Comment by mikelhamer** [comment]
Thanks as always for the great explanation and possible solution!

**Comment by mikelhamer** [comment]
Awesome! Could you please briefly explain why the Copilot default is 60s rather than the ECS default of 300s?

**Comment by bvtujo** [comment]
We want to strike a balance between "snappy deployments" and "time enough for your containers to come up," and we felt that the default of 5 minutes was far too long for the majority of modern workloads. The grace period adds additional time to deployments, as the LB has to wait the entire duration of the grace period before the target can be considered healthy. 

Similar reasoning for the deregistration delay default, as when redeploying a service the LB has to wait the entire duration of the deregistration delay and we don't want to add an extra 5 minutes to every service update.

**Comment by Lou1415926** [comment]
The feature is now released in [v1.9.0](https://github.com/aws/copilot-cli/releases/tag/v1.9.0)! 🎊

**Comment by mikelhamer** [comment]
Yay! Thanks!!
